src/OrchardCore/OrchardCore.Apis.GraphQL.Abstractions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,51 @@
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Apis.GraphQL.Queries;
+using OrchardCore.Apis.GraphQL.Types;
+
+namespace OrchardCore.Apis
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void AddGraphMutationType<T>(this IServiceCollection services) where T : MutationFieldType
+        {
+            services.AddScoped<T>();
+            services.AddScoped<MutationFieldType, T>();
+        }
+
+        public static void AddGraphQueryType<T>(this IServiceCollection services) where T : QueryFieldType
+        {
+            services.AddScoped<T>();
+            services.AddScoped<QueryFieldType, T>();
+        }
+
+        public static void AddGraphQLInputType<TContenPart, TInputType>(this IServiceCollection services) 
+            where TContenPart : class 
+            where TInputType : InputObjectGraphType<TContenPart>
+        {
```

Comment:
Probably we should rename TContenPart to something else, cause `OrchardCore.Apis.GraphQL.Abstractions` project, should not be CMS specific

Reply:
Agreed! - Resolved